File: source_formatting/hebrew_small_to_emphasis.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

from source_formatting.html_parser import BaseHtmlTranslator

class HebrewSmallToEmphasisTagTranslator(BaseHtmlTranslator):
    def handle_starttag(self, tag, attrs):
        if tag == "small":
            tag = "span"
            attrs = [("class", "hebrew-emphasis")]
        self.append_start_tag(tag, attrs)

    def handle_endtag(self, tag):
        if tag == "small":
            tag = "span"
        self.append_end_tag(tag)

    def handle_data(self, data):
        self._out.append(data)

File: source_formatting/hebrew_small_to_emphasis.ts
import {Attributes, HtmlVisitor} from "./html_visitor";

export class HebrewSmallToEmphasisTagTranslator extends HtmlVisitor {
  shouldRun(input: string): boolean {
    return input.includes("<small");
  }

  visitStartTag(tag: string, attributes: Attributes): void {
    if (tag === "small") {
      this.appendStartTag("span", [["class", "hebrew-emphasis"]]);
    } else {
      this.appendStartTag(tag, attributes);
    }
  }

  visitEndTag(tag: string): void {
    this.appendEndTag(tag === "small" ? "span" : tag);
  }

  visitText(text: string): void {
    this.appendText(text);
  }
}
